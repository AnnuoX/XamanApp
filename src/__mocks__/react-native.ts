/* eslint-disable implicit-arrow-linebreak */
/* eslint-disable spellcheck/spell-checker */

const ReactNative = require('react-native');
const crypto = require('crypto');

ReactNative.NativeModules.CryptoModule = {
    randomKeySync: jest.fn((len: number) =>
        crypto
            .randomBytes(len)
            .toString('hex')
            .toUpperCase(),
    ),
    randomKey: jest.fn((len: number) =>
        Promise.resolve(
            crypto
                .randomBytes(len)
                .toString('hex')
                .toUpperCase(),
        ),
    ),
    sha512: jest.fn(value =>
        Promise.resolve(
            crypto
                .createHash('sha512')
                .update(value)
                .digest('hex'),
        ),
    ),
    sha256: jest.fn(value =>
        Promise.resolve(
            crypto
                .createHash('sha256')
                .update(value)
                .digest('hex'),
        ),
    ),
    sha1: jest.fn(value =>
        Promise.resolve(
            crypto
                .createHash('sha1')
                .update(value)
                .digest('hex'),
        ),
    ),
    hmac256: jest.fn((value, key) =>
        Promise.resolve(
            crypto
                .createHmac('sha256', key)
                .update(value)
                .digest('hex'),
        ),
    ),
    encrypt: jest.fn((entry, key) => {
        const ivHex = crypto
            .randomBytes(16)
            .toString('hex')
            .toUpperCase();
        const keyEnc = crypto
            .createHash('sha256')
            .update(key)
            .digest();

        const iv = Buffer.from(ivHex, 'hex');
        const cipher = crypto.createCipheriv('aes-256-cbc', keyEnc, iv);
        entry = Buffer.from(entry);
        let crypted = cipher.update(entry, 'utf-8', 'base64');
        crypted += cipher.final('base64');
        return Promise.resolve({ cipher: crypted, iv: ivHex });
    }),
    decrypt: jest.fn((entry, key, iv) => {
        const keyEnc = crypto
            .createHash('sha256')
            .update(key)
            .digest();
        iv = Buffer.from(iv, 'hex');
        const decipher = crypto.createDecipheriv('aes-256-cbc', keyEnc, iv);
        let decrypted = decipher.update(entry, 'base64');
        decrypted += decipher.final();
        return Promise.resolve(decrypted);
    }),
};

module.exports = ReactNative;
