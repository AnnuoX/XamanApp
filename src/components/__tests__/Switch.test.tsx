/**
 * @format
 */

import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import { Switch as RNSwitch } from 'react-native';

import { Switch } from '..';

describe('[Switch]', () => {
    it('renders correctly', () => {
        const onChange = jest.fn();
        const tree = renderer.create(<Switch checked onChange={onChange} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders with given label', () => {
        const label = 'A Label I Am';
        const onChange = jest.fn();
        const tree = renderer.create(<Switch title={label} checked onChange={onChange} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('calls the given function when the switch is pressed', () => {
        const onChange = jest.fn();
        const tree = renderer.create(<Switch checked onChange={onChange} />);

        const switchInst = tree.root.findByType(RNSwitch);

        switchInst.props.onValueChange();

        expect(onChange).toHaveBeenCalled();
    });
});
