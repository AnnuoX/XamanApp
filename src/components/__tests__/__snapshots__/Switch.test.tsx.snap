// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Switch] renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingBottom": 5,
      "paddingTop": 5,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 32.14285714285714,
        "marginLeft": 10,
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "center",
      }
    }
  >
    <View>
      <RCTSwitch
        accessibilityRole="button"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "height": 31,
              "width": 51,
            },
            Object {
              "alignSelf": "center",
            },
          ]
        }
        value={true}
      />
    </View>
  </View>
</View>
`;

exports[`[Switch] renders with given label 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingBottom": 5,
      "paddingTop": 5,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 32.14285714285714,
        "marginLeft": 10,
      }
    }
  >
    A Label I Am
  </Text>
  <View
    style={
      Object {
        "alignSelf": "center",
      }
    }
  >
    <View>
      <RCTSwitch
        accessibilityRole="button"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "height": 31,
              "width": 51,
            },
            Object {
              "alignSelf": "center",
            },
          ]
        }
        value={true}
      />
    </View>
  </View>
</View>
`;
